examples/common_spaces.jl

```diff
@@ -1,87 +1,101 @@
 using ClimaCore: Geometry, Domains, Meshes, Topologies, Spaces
 
-abstract type Space{FT} end
+abstract type MeshDescription{FT} end
 
-Base.@kwdef struct PeriodicLine{FT} <: Space{FT}
-    xmax::FT
-    xelem::Int
-    npoly::Int
+Base.@kwdef struct PeriodicLine{FT} <: MeshDescription{FT}
+    x_max::FT
+    x_elem::Int
 end
 
-Base.@kwdef struct PeriodicRectangle{FT} <: Space{FT}
-    xmax::FT
-    ymax::FT
-    xelem::Int
-    yelem::Int
-    npoly::Int
+Base.@kwdef struct PeriodicRectangle{FT} <: MeshDescription{FT}
+    x_max::FT
+    y_max::FT
+    x_elem::Int
+    y_elem::Int
 end
 
-Base.@kwdef struct CubedSphere{FT} <: Space{FT}
+Base.@kwdef struct CubedSphere{FT} <: MeshDescription{FT}
```

Comment:
Can we not use struct names like CubedSphere, etc. There is potential for confusion with structs defined in src/Meshes. Can we replace them with default arguments to make_space / make_mesh functions!

Reply:
How about I replace each struct with function, so `h_mesh = periodic_line_mesh(; x_max, x_elem)`, etc.?